python/lsst/pipe/tasks/extended_psf.py

```diff
@@ -463,29 +503,44 @@ def select_detector_refs(self, ref_list):
                 `lsst.daf.butler._deferredDatasetHandle.DeferredDatasetHandle`
             List of available bright star stamps data references.
         """
-        region_ref_list = {region: [] for region in self.focal_plane_regions.keys()}
+        region_ref_list = {region: [] for region in self.detectors_focal_plane_regions.keys()}
         for dataset_handle in ref_list:
-            det_id = dataset_handle.ref.dataId["detector"]
-            if det_id in self.regionless_dets:
+            detector_id = dataset_handle.ref.dataId["detector"]
+            if detector_id in self.regionless_dets:
                 continue
             try:
-                region_name = self.config.detectors_focal_plane_regions[det_id]
+                region_name = find_region_for_detector(detector_id, self.detectors_focal_plane_regions)
             except KeyError:
                 self.log.warning(
                     "Bright stars were available for detector %d, but it was missing from the %s config "
                     "field, so they will not be used to build any of the extended PSF models.",
-                    det_id,
+                    detector_id,
                     "'detectors_focal_plane_regions'",
                 )
-                self.regionless_dets.append(det_id)
+                self.regionless_dets.append(detector_id)
```

Comment:
Are we expecting the code to execute this `except` block frequently, if only a few detectors are assigned to regions? Is the typical use case expected to have all (at least most) detectors assigned to regions, or only for a handful number of detectors?

Reply:
This `except` will only trigger if a given detector was not specified by the user in any of the specified regions. This should hopefully be a rare case - i.e., if I'm splitting a visit into regions, I should hope to specify all detectors. The obvious warning should also be useful in this case, in case this is an oversight by the end-user.